File: 852.Peak Index in a Mountain Array/_852.py
class Solution:
    def peakIndexInMountainArray(self, A:) -> int:
        start = 0
        end = len(A)-1
        while start<=end:
            mid = (start+end)//2
            if A[mid]>A[mid-1] and A[mid]>A[mid+1]:
                return mid
            if A[mid]>A[mid-1]:
                start = mid+1
            else:
                end = mid-1

File: 852.Peak Index in a Mountain Array/852.cpp
class Solution {
public:
    int peakIndexInMountainArray(vector<int>& A) {
        int end = A.size()-1;
		int start = 0;
		while(start<=end){
			int mid = (start+end)/2;
			if(A[mid]>A[mid+1] && A[mid]>A[mid-1]){
				return mid;
			}
			if (A[mid]>A[mid-1]){
				start = mid+1;
			}else{
				end = mid-1;
			}
		}
		return 0;
    }
};